From d0790bdbf8559364a2057b4bf7424778fce10350 Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Mon, 20 May 2024 13:50:06 +0200
Subject: [PATCH 0864/1695] dts: stm32: add mlahb node on stm32mp21

Add mlahb node for stm32mp21x platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1d0d9a63f0849af1137805807f5bc75e029bd0cb dts: stm32: add mlahb node on stm32mp21]
Change-Id: I99109ff0b7c05138fc927ffd70fcef92f1f1a49d
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/383219
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 1be2f4a42..16f4b6e56 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -112,6 +112,23 @@
 		};
 	};
 
+	mlahb: ahb@1 {
+		compatible = "st,mlahb", "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges = <0x0 0x0  0xFFFFFFFC>;
+		dma-ranges = <0x0 0x0 0xFFFFFFFC>;
+
+		m33_rproc: m33@0 {
+			compatible = "st,stm32mp2-m33";
+			reg = <0 0>;
+			resets = <&rcc C2_R>,
+				 <&rcc C2_HOLDBOOT_R>;
+			reset-names = "mcu_rst", "hold_boot";
+			status = "disabled";
+		};
+	};
+
 	scmi_regu: scmi-regu {
 		compatible = "st,scmi-regulator-consumer";
 		#address-cells = <1>;
-- 
2.47.2

